pkg/resources/fluentd/drainjob.go

```diff
@@ -0,0 +1,102 @@
+// Copyright © 2021 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fluentd
+
+import (
+	"strings"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func (r *Reconciler) drainJobFor(pvc corev1.PersistentVolumeClaim) (*batchv1.Job, error) {
+	bufVolName := r.Logging.QualifiedName(r.Logging.Spec.FluentdSpec.BufferStorageVolume.PersistentVolumeClaim.PersistentVolumeSource.ClaimName)
+
+	var initContainers []corev1.Container
+	if c := r.volumeMountHackContainer(); c != nil {
+		initContainers = append(initContainers, *c)
+	}
+
+	fluentdContainer := r.fluentContainer() // TODO: don't redirect container logs
+	fluentdContainer.VolumeMounts = append(fluentdContainer.VolumeMounts, corev1.VolumeMount{
+		Name:      bufVolName,
+		MountPath: bufferPath,
+	})
+	containers := []corev1.Container{
+		*fluentdContainer,
+		r.drainWatchContainer(bufVolName),
+	}
+	if c := r.bufferMetricsSidecarContainer(); c != nil {
+		containers = append(containers, *c)
+	}
+
+	spec := batchv1.JobSpec{
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: r.generatePodMeta(),
```

Comment:
please modify this call to allow passing the component name so that the drainer job's pods won't get traffic from the service

Reply:
Instead, I used only the relevant part from `generatePodMeta`.